### ISSUE TITLE
Deduplication on Rust is lacking

### ISSUE THREAD

**Issue by DavidKorczynski** [google/oss-fuzz#7040]
Recently on the json5format a lot of issues were reported but they all seem to be related to the same. The monorail table looks as follows:

```
star_border | 42641 | Bug | ---- | New | json5format | 2021-12-19 | ---- | json5format:fuzz_parse: ASSERT: byte index 9 is not a char boundary; it is inside ClusterFuzz Reproducible | link
star_border | 42642 | Bug | ---- | New | json5format | 2021-12-19 | ---- | json5format:fuzz_parse: ASSERT: byte index NUMBER is not a char boundary; it is inside ClusterFuzz Reproducible | link
star_border | 42646 | Bug | ---- | New | json5format | 2021-12-19 | ---- | json5format:fuzz_parse: ASSERT: byte index 326 is not a char boundary; it is inside ClusterFuzz Reproducible | link
star_border | 42649 | Bug | ---- | New | json5format | 2021-12-19 | ---- | json5format:fuzz_parse: ASSERT: byte index 24 is not a char boundary; it is inside ClusterFuzz Reproducible | link
star_border | 42659 | Bug | ---- | New | json5format | 2021-12-19 | ---- | json5format:fuzz_parse: ASSERT: assertion failed: self.scope_stack.len() > 0 ClusterFuzz Reproducible | link
star_border | 42669 | Bug | ---- | New | json5format | 2021-12-19 | ---- | json5format:fuzz_parse: ASSERT: byte index 4 is not a char boundary; it is inside ClusterFuzz Reproducible | link
star_border | 42675 | Bug | ---- | New | json5format | 2021-12-19 | ---- | json5format:fuzz_parse: ASSERT: byte index 735 is not a char boundary; it is inside ClusterFuzz Reproducible | link
star_border | 42676 | Bug | ---- | New | json5format | 2021-12-19 | ---- | json5format:fuzz_parse: ASSERT: byte index 34 is not a char boundary; it is inside ClusterFuzz Reproducible | link
star_border | 42573 | Bug | ---- | New | json5format | 2021-12-18 | ---- | json5format:fuzz_parse: ASSERT: byte index 697 is not a char boundary; it is inside ClusterFuzz Reproducible | link
star_border | 42580 | Bug | ---- | New | json5format | 2021-12-18 | ---- | json5format:fuzz_parse: ASSERT: byte index 5 is not a char boundary; it is inside ClusterFuzz Reproducible | link
star_border | 42584 | Bug | ---- | New | json5format | 2021-12-18 | ---- | json5format:fuzz_parse: ASSERT: byte index 96 is not a char boundary; it is inside ClusterFuzz Reproducible | link
star_border | 42585 | Bug | ---- | New | json5format | 2021-12-18 | ---- | json5format:fuzz_parse: ASSERT: byte index 29 is not a char boundary; it is inside ClusterFuzz Reproducible
...
...
```

Looking at the stack traces it's clear they originate from the same function, but it seems that a bug is filed for each byte index that is not a char.

There's about 85 of issues of this nature, and they all seem to be the same.

**Comment by DavidKorczynski** [comment]
Monorail query https://bugs.chromium.org/p/oss-fuzz/issues/list?q=%22ASSERT%3A%20byte%20index%22%20json5format&can=1

**Comment by oliverchang** [comment]
This should be fixed now as part of the mitigations in #2533. Please let us know if you notice this again!
